x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetDeploymentStatsAction.java

```diff
@@ -230,15 +231,21 @@ public int hashCode() {
             private AllocationStatus allocationStatus;
             private String reason;
             private final ByteSizeValue modelSize;
+            @Nullable private final Integer inferenceThreads;
+            @Nullable private final Integer modelThreads;
```

Comment:
Why are these nullable? It might be good to return the default value when the user didn't set them.

Reply:
I was in a dilemma on that one. I think it is possible in the future we will develop a smart default strategy. I prefer to keep them `nullable` for now, so that if we do get there, running deployments can benefit. If we decide we won't we can always make them non-nullable, but the other way around is harder.

Reply:
> running deployments can benefit.

This would be equally tricky though. 

If a deployment was started with defaults, then they do a rolling upgrade, now magically, without intervention, their deployment is using more resources. That seems dangerous to me.

Reply:
> Also, is it even possible to update the threads used once a process has started?

If the cluster is upgraded, then the nodes will end up relaunching the process eventually.

Reply:
After discussion offline, we agreed that if smart allocation/threading is developed in the future, the users can benefit by restarting an allocation. I'll change the PR to explicitly set the defaults.